### ISSUE TITLE
Try automatic resolution based on path and not load from the path

### ISSUE THREAD

**Issue by SarasArya** [neverendingqs/serverless-dotenv-plugin#33]
So I have this monorepo setup where I want to have my .env.development and .env.production at the root level and reference the required variables in my service folders.
Right now the path variable tries to load it directly from the file, without trying to find the file. 

The issue with this approach is that if I hardcode what is my .env right now. I might have to change the path name in each service folder to production and then deploy. Which is cumbersome and feels wrong.

What I would like an option as `readFrom` and given a path and the whole thing will run on that path instead of the current repo. 

What do you guys think around this? Anybody faced same issue? I Might want to put in a PR on this but need suggestions from collaborators, also is there anyway we can currently do this that I might be missing?

**Comment by colynb** [comment]
Hi SarasArya, I'm having trouble wrapping my head around what the issue is. Is there any way you can point me to an example project or record a screencast, or something more visual?

**Comment by bryankennedy** [comment]
I _think_ I have a similar issue.

I run my serverless code inside of a subfolder of my larger project and the .env.${NODE_ENV} file is at the root. So in my server package I have this script:
```
NODE_ENV=development serverless deploy
```

Within my serverless.yml with the dotenv plugin I have:
```
custom:
  dotenv:
    path: ../../.env.${env:NODE_ENV}
```
But that doesn't work. The system doesn't load any of the env vars in that file.

If I hard code it like:
```
custom:
  dotenv:
    path: ../../.env.development
```
It works fine.

Is it possible to use environment variables in the setup of dotenv before it loads them from the file? I hope that makes a bit of sense. @SarasArya I hope this is the same as what you're trying to accomplish.

**Comment by webchaz** [comment]
@bryankennedy Were you able to find a solution to this? I'm in the same boat.

**Comment by bryankennedy** [comment]
Unfortunately, I'm just working around this by having multiple .env files for now. This allows me to just directly reference the .env file without trying to go outside of the containing directory. Not ideal, but it's all that I could get working for now.

**Comment by colynb** [comment]
@bryankennedy I haven't looked at the code in a long time but if you take a look at the source code you see that `.env.[NODE_ENV]` will get loaded automatically. There's no need to include that in your config. Let me know if I'm missing something.

**Comment by SarasArya** [comment]
Hey @bryankennedy My issue was the same. Also, there should be an option on where is my .env file. It assumes it's in my root of the service. But what if it's not. What if it's in the root of the project. Hence the `readFrom` option

**Comment by webchaz** [comment]
@colynb It's just this line here: https://github.com/colynb/serverless-dotenv-plugin/blob/master/index.js#L33

Since it's just returning the path, it doesn't support '.production', '.development', etc.  It needs the fully resolved path + name.  If this was updated to be the same as the next return, only with path prepended should be all good.

**Comment by colynb** [comment]
> Since it's just returning the path, it doesn't support '.production', '.development', etc. It needs the fully resolved path + name. If this was updated to be the same as the next return, only with path prepended should be all good.

@webchaz I see. Because the path is custom, you lose the ability to use the environment. If a lot of people are using the path config option to set the full path instead of the base path, I make an update as you suggested, it would be a breaking change I think. I might have to include another option like "basePath".

**Comment by webchaz** [comment]
@colynb that would be great or maybe a new major version to keep the api the same - totally up to you, either way would be awesome!

**Comment by webchaz** [comment]
@colynb This should handle it:

```
resolveEnvFileName(env)  {
  const configPath = (this.config && this.config.path) ? this.config.path : ''
  return fs.existsSync(`${configPath}.env.${env}`) ? configPath + '.env.' + env : '.env'
}
```

**Comment by colynb** [comment]
Ok, I just published a new version, 2.1.0. It includes a `basePath` option now.

**Comment by webchaz** [comment]
@colynb Awesome, just tried it out, works great - thank you!
